diag_manager/fms_diag_yaml.F90

```diff
@@ -1,62 +1,324 @@
+  if (trim(field%string_var_write) .eq. "false") field%var_write = .false.
+
+  call diag_get_value_from_key(file_id, var_id, "output_name", field%var_outname)
+  call diag_get_value_from_key(file_id, var_id, "long_name", field%var_longname, is_optional=.true.)
+  !! VAR_UNITS !!
+
+  natt = 0
+  natt = get_num_blocks(file_id, "attributes", parent_block_id=var_id)
+  if (natt .ne. 0) then
+    call get_block_ids(file_id, "attributes", var_att_id, parent_block_id=var_id)
+    nkeys = get_nkeys(file_id, var_att_id(1))
+    allocate(key_ids(nkeys))
+    call get_key_ids(file_id, var_att_id(1), key_ids)
+
+    allocate(field%var_attributes(nkeys, 2))
+    do j = 1, nkeys
+      call get_key_name(file_id,  key_ids(j), field%var_attributes(j, 1))
+      call get_key_value(file_id, key_ids(j), field%var_attributes(j, 2))
+    enddo
+    deallocate(key_ids)
+  endif
+
+end subroutine
+
+!< @brief diag_manager wrapper to get_value_from_key to use for allocatable
+!! string variables
+subroutine diag_get_value_from_key(file_id, par_id, key_name, value_name, is_optional)
+  integer,           intent(in) :: file_id     !< Id of the diag_table yaml file
+  integer,           intent(in) :: par_id      !< Id of the parent block in the yaml file
+  character(len=*),  intent(in) :: key_name    !< Key to look for in the parent block
+  character(len=:), allocatable :: value_name  !< Value of the key
+  logical, intent(in), optional :: is_optional !< Flag indicating if the key is optional
+
+  character(len=255) :: buffer !< String buffer to read in to
+
+  call get_value_from_key(file_id, par_id, trim(key_name), buffer, is_optional= is_optional)
+  allocate(character(len=len_trim(buffer)) :: value_name)
+  value_name = trim(buffer)
+
+end subroutine diag_get_value_from_key
+
+!< @brief gets the lat/lon of the sub region to use in a diag_table yaml
+subroutine get_region_latlon(diag_yaml_id, region_id, lat_lon_region)
+  integer, intent(in)  :: diag_yaml_id       !< Id of the diag_table yaml file
+  integer, intent(in)  :: region_id          !< Id of the region block to read from
+  real,    intent(out) :: lat_lon_region (8) !< sub region
+
+  call get_value_from_key(diag_yaml_id, region_id, "xbegin", lat_lon_region(1), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "xend", lat_lon_region(2), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "ybegin", lat_lon_region(3), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "yend", lat_lon_region(4), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "zbegin", lat_lon_region(5), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "zend", lat_lon_region(6), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "ubegin", lat_lon_region(7), is_optional=.true.)
+  call get_value_from_key(diag_yaml_id, region_id, "uend", lat_lon_region(8), is_optional=.true.)
```

Comment:
What is `u`?  Wind?

Reply:
u is suppose to be the fourth dimension (i.e spectral dimension, species or whatever). I wasn't sure what naming convention to use here.

Reply:
Maybe instead of "{x/y/z/u}_begin" and "{x/y/z/u}_end", we can do: "dim{1,2,3,4}_begin" and "dim{1,2,3,4}_end" which is more general?

Reply:
I think the more general approach is a good idea.  If you document what u is, or name that `other_dim` or something like that, it would also be fine.  Or you could document that `u` is a utility 4th dimension.  Either way is up to you.